### PR TITLE
[refactor](planner) refactor automatically set instance_num

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/system/Backend.java [hunk 1]
```diff
@@ -519,6 +542,24 @@ public void updateDisks(Map<String, TDisk> backendDisks) {
...
+            isChanged = true;
+        }
+        if (this.pipelineExecutorSize != pipelineExecutorSize) {
+            this.pipelineExecutorSize = pipelineExecutorSize;
+            isChanged = true;
+        }
+        if (isChanged) {
+            LOG.info("add new cpu info. backendId: {}, cpucores: {}, pipelineExecutorSize: {}", id, cpuCores,
+                    pipelineExecutorSize);
+            Env.getCurrentEnv().getEditLog().logBackendStateChange(this);
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/system/Backend.java:559]: 不要在这加这个逻辑

File: fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java [hunk 2]
```diff
@@ -256,7 +256,7 @@ protected void exec() {
                 ReportHandler.taskReport(beId, tasks);
             }
             if (disks != null) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java:258]: 不要使用disk report了，单独写一个 cpuInfoReport

File: fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java [hunk 3]
```diff
@@ -549,16 +549,17 @@ private static void taskReport(long backendId, Map<TTaskType, Set<Long>> running
...
         LOG.info("begin to handle disk report from backend {}", backendId);
         long start = System.currentTimeMillis();
         Backend backend = Env.getCurrentSystemInfo().getBackend(backendId);
         if (backend == null) {
             LOG.warn("backend doesn't exist. id: " + backendId);
             return;
         }
-
         backend.updateDisks(backendDisks);
+        backend.updateCpuInfo(cpuCores, pipelineExecutorSize);
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java:562]: 在这里判断cpu或者pipeline 配置是否有变更，如果变更了，写system info

**Comment by Mryange** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 4]
```diff
@@ -1433,8 +1433,8 @@ public void setEnableFoldConstantByBe(boolean foldConstantByBe) {
 
     public int getParallelExecInstanceNum() {
         if (enablePipelineEngine && parallelPipelineTaskNum == 0) {
-            Backend.BeInfoCollector beinfoCollector = Backend.getBeInfoCollector();
-            return beinfoCollector.getParallelExecInstanceNum();
+            int size = Env.getCurrentSystemInfo().getMinPipelineExecutorSize();
+            return (size + 1) / 2;
```
Comment (BiteTheDDDDt) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1437]: Is it possible that size is int_max?

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**